pkg/network/connector.go

```diff
@@ -0,0 +1,44 @@
+package network
+
+import (
+	"encoding/json"
+
+	"github.com/CESARBR/knot-babeltower/pkg/logging"
+)
+
+const (
+	exchangeConnIn = "connIn"
+	registerInKey  = "device.register"
+)
+
+type connector struct {
+	logger logging.Logger
+	amqp   *Amqp
+}
+
+// Connector handle messages received from a service
+type Connector interface {
+	SendRegisterDevice(string, string) error
+}
+
+// NewConnector constructs the Connector
+func NewConnector(logger logging.Logger, amqp *Amqp) Connector {
+	return &connector{logger, amqp}
+}
+
+// SendRegisterDevice sends a registered message
+func (mp *connector) SendRegisterDevice(id string, name string) error {
+	mp.logger.Debug("Sending register message")
+
+	msg := struct {
```

Comment:
Do we have any entity that represents this `user` register structure in another file? If so, we can turn it into a shared entity so that it can be used here.

Reply:
Done